dask_deltatable/core.py

```diff
@@ -43,13 +43,16 @@ def __init__(
         self.fs, self.fs_token, _ = get_fs_token_paths(
             path, storage_options=storage_options
         )
-        self.schema = self.dt.pyarrow_schema()
+        self.schema = self.dt.schema().to_pyarrow()
 
     def read_delta_dataset(self, f: str, **kwargs: Dict[any, any]):
         schema = kwargs.pop("schema", None) or self.schema
         filter = kwargs.pop("filter", None)
         if filter:
-            filter_expression = pq._filters_to_expression(filter)
+            filter_converter = getattr(
+                pq, "filters_to_expression", getattr(pq, "_filters_to_expression")
+            )
+            filter_expression = filter_converter(filter)
```

Comment:
I'm guessing this is to handle a private method that has been made public in pyarrow?

I wonder if it would feel more pythonic and readable to use a try/except here and ask forgiveness not permission?

```suggestion
            try:
                filter_expression = pq._filters_to_expression(filter)
            except AttributeError:
                filter_expression = pq.filters_to_expression(filter)
```

Reply:
Good point, changed.